### PR TITLE
Hello Data!

### PR DESCRIPTION
**Pull request by ActuallyTaylor**

### CODE REVIEW

File: include/charge_controller/dev/LCDDisplay.hpp [hunk 1]
```diff
@@ -2,24 +2,156 @@
...
+     */
+    LCDDisplay(IO::GPIO& reg_select, IO::GPIO& reset, IO::SPI& spi);
 
-  void showSplashScreen();
-  void show();
+    /**
+     * Initializes the LCD driver and displays the splash image.
+     */
+    void init();
+    /**
```
Comment (mjmagee991) [inline include/charge_controller/dev/LCDDisplay.hpp:27]: Add newlines between the functions

File: include/charge_controller/dev/LED.hpp [hunk 2]
```diff
@@ -5,39 +5,39 @@
...
-   */
-  void setState(EVT::core::IO::GPIO::State state);
-
-private:
-  /// The gpio pin used by the LED
-  EVT::core::IO::GPIO &gpio;
-  /// If the LED is active high or active low
-  ActiveState activeState;
-};
+    class LED {
```
Comment (mjmagee991) [inline include/charge_controller/dev/LED.hpp]: Code inside namespaces shouldn't be indented

File: src/charge_controller/dev/LCDDisplay.cpp [hunk 3]
```diff
@@ -1,6 +1,70 @@
...
+
+void LCDDisplay::display() {
+    lcd.setTextForSection(0, batteryOneStatus);
+
+    lcd.setTextForSection(1, chargeControllerStatus);
+
+    lcd.setTextForSection(2, batteryTwoStatus);
+
+    char batteryOneVolt[32];
+    std::sprintf(batteryOneVolt, "%d v", batteryOneVoltage);
```
Comment (mjmagee991) [inline src/charge_controller/dev/LCDDisplay.cpp]: Volts use a capital V
This is quite possibly one of the nitpickiest comments I have ever left

File: src/charge_controller/dev/LCDDisplay.cpp [hunk 4]
```diff
@@ -1,6 +1,70 @@
...
+    lcd.setTextForSection(9, bat1MaxTemp);
+
+    char currentDisplay[32];
+    std::sprintf(currentDisplay, "%d A", chargeControllerCurrent);
+    lcd.setTextForSection(10, currentDisplay);
+
+    char bat2MaxTemp[32];
+    std::sprintf(bat2MaxTemp, "%d C", batteryTwoMaxTemp);
+    lcd.setTextForSection(11, bat2MaxTemp);
+}
```
Comment (mjmagee991) [inline src/charge_controller/dev/LCDDisplay.cpp]: Newline

**Review by mjmagee991 — CHANGES_REQUESTED** [review]
A couple minor issues. Also, don't worry about reformatting anything. Once we get the autoformatter set up for this repo, most of it will get cleaned up automatically

**Comment by ActuallyTaylor** [comment]
[Charge Controller LCD](https://trello.com/c/xwewglwV/46-charge-controller-lcd)

**Review by mjmagee991 — APPROVED** [review]
Nice work

**Review by S-TH3TA-13 — APPROVED** [review]

**Pull request merged**